### DISCUSSION TITLE
Relationship dot notation not working

### DISCUSSION THREAD

**Discussion by repeater09** [filamentphp/filament#4826] [Help]
In `User.php` I have a simple relationship:

```
    public function car()
    {
        return $this->hasOne(Car::class, 'CarID', 'car_id');
    }
```

in `UserResource.php` I have:

`Tables\Columns\TextColumn::make('car.Model'),`

However, it's blank. I can use debugbar and see the query is correct.

I can also do:

`Tables\Columns\TextColumn::make('car'),`

and get the object from that relationship displayed on the table. 
```
{"CarID":1, "Model":"Ford"}
```

Any ideas? Feel like I am doing something wrong but I can see the data with the column I am wanting to use.

**Comment by RX-77IT** [comment]
Is the third parameter in the relationship correct? Shouldn't it be the [local key of the User class](https://laravel.com/docs/9.x/eloquent-relationships#one-to-one)?

What happens if you remove that third parameter?

**Reply by repeater09** [reply]
I changed the data up just for example purposes. Like I said the data is being returned. But if I use the dot notation it doesn't show. But if I just call the relationship it shows.

**Reply by RX-77IT** [reply]
Very strange. I did a quick test and it works. I really don't know what could be blocking the dot notation.

... maybe it's the capital letters of the columns?

**Reply by repeater09** [reply]
> Very strange. I did a quick test and it works. I really don't know what could be blocking the dot notation.
> 
> ... maybe it's the capital letters of the columns?

I was thinking that too. I am going to test that out and see.

This is an existing database that I am having to work with. It's not ideal by any means and I am worried I will have to go a different route to support some of the weirder things.

**Reply by repeater09** [reply]
> Very strange. I did a quick test and it works. I really don't know what could be blocking the dot notation.
> 
> ... maybe it's the capital letters of the columns?

Yeah. So I verified I can show a relationship with a column that is lowercase. But not capitalized. 

I wonder if there are any ideas that may help.

**Reply by mrozek90** [reply]
> Yeah. So I verified I can show a relationship with a column that is lowercase. But not capitalized.

I not able to show any relationship in table via dot notation, but forms relationship working correctly...

Form working:
` Forms\Components\Select::make('cities_id')
                    ->relationship('city', 'name')`
Table not:
`Tables\Columns\TextColumn::make('city.name')`

Did you solve that problem?

**Reply by mrozek90** [reply]
Ok i solve it myself. :-)
Need to pass second parameter function **belongsTo($related, $foreginKey)** in Model Class.


Example:
```
public function city(): BelongsTo
    {
        return $this->belongsTo(City::class, 'cities_id');
    }
```
